Assets/Scripts/GameScene/PlayerController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerController : MonoBehaviour, IPlayerController
+{
+    [SerializeField] private Board board;
+    private PlayerInput _playerInput;
+    private CharacterController _characterController;
+    
+    
+    private Vector2 _movementInput;
+    private Vector3 _playerMovement;
+    private float _playerSpeed = 2.0f;
+
+    
+    private bool _isMovementPressed;
+    [SerializeField] private float rotationFactorPerFrame = 1.0f;
+    
+    
+    private void Awake()
+    {
+        InitPlayerInput();
+        _characterController = GetComponent<CharacterController>();
```

Comment:
Let's try to refrain from using `GetComponent` as it creates a dependency between the script and the GameObject it's attached to, make `_characterController` a `[SerializeField] private...`